fabric-biomes-v1/src/main/java/net/fabricmc/fabric/impl/biome/InternalBiomeUtils.java

```diff
@@ -148,7 +158,22 @@ public static void injectBiomesIntoClimate(LayerRandomnessSource random, int[] v
 		}
 	}
 
-	public static void recreateChunkGenerators(GeneratorOptions generatorOptions) {
-		((DimensionOptionsAccessor) (Object) generatorOptions.getDimensionMap().get(DimensionOptions.NETHER)).setChunkGenerator(DimensionTypeAccessor.createNetherGenerator(generatorOptions.getSeed()));
+	public static int getRawId(RegistryKey<Biome> key) {
+		return BuiltinRegistries.BIOME.getRawId(BuiltinRegistries.BIOME.method_31140(key));
```

Comment:
I would suggest capturing the `biomeRegistry` in `VanillaLayeredBiomeSource.<init>` and use that one theoughout for the conversion `RegistryKey <-> RawId`. That should be compatible with using the DynamicRegistryManager then (and I think also json biomes).
We'd have to check if that capture covers all use cases, but I think it does.

Reply:
This will still not fix the Vanilla logic that also accesses the raw-id mapping directly and thus just complicates the logic needlessly.